### DISCUSSION TITLE
Support PEG syntax as an alternative input format?

### DISCUSSION THREAD

**Discussion by ethindp** [tree-sitter/tree-sitter#3514] [Ideas]
### Problem

I have a large PEG grammar (from PackCC) that I want to use Tree-Sitter for instead. The problem is that the grammar is quite large, and rewriting it into TS's javascript-ish syntax is cumbersome.

### Expected behavior

Ideally, it would be nice if tree-sitter had a convert tool that could accept a few different input formats (maybe starting with PEG) that could take an existing grammar and rewrite it in Tree-Sitter's format.

**Comment by ethindp** [comment]
I should note that this tool wouldn't handle ambiguities or precedence of operators or anything like that, it would just be a direct translation. You would still need to edit the resulting grammar to fix it's problems if it had any.

**Comment by mingodad** [comment]
Probably this can help you !

I've added several non trivial grammars Javascript, Java, Kotlin, PHP, C, CPP, Rust, Ruby, CSS, HTML, Python using a quickjs script to convert "src/grammar.json" to an EBNF understood by https://mingodad.github.io/lua-wasm-playground/ (the script is here https://github.com/mingodad/plgh/blob/main/json2ebnf-lua.js).
